tests/e2e/plugins/e2e-plan-data-interceptor.php

```diff
@@ -20,14 +20,19 @@
 function e2e_intercept_plan_data_request( $return, $r, $url ) {
 	$site_id = Jetpack_Options::get_option( 'id' );
 
-	// shortcut the api call...
-	if ( false !== stripos( $url, sprintf( '/sites/%d?', $site_id ) ) ) {
+	if ( empty( $site_id ) ) {
+		return $return;
+	}
 
+	// match both /sites/$site_id && /sites/$site_id? urls
+	if ( 1 === preg_match( sprintf( '/\/sites\/%d($|\?)/', $site_id ), $url ) ) {
 		$plan_data = get_option( 'e2e_jetpack_plan_data' );
 		if ( empty( $plan_data ) ) {
 			return $return;
 		}
 
+		delete_option( 'jetpack_active_plan' );
```

Comment:
Not sure if we need to delete the option here, but I thought it could prevent some of the race conditions I was experiencing, which might be related to this chunk:

https://github.com/Automattic/jetpack/blob/try/e2e-plan-data-stability/class.jetpack-plan.php#L54-L59

Better safe than sorry 🤷‍♂